src/core/context/AppContextProvider.tsx

```diff
@@ -528,10 +528,13 @@ async function callTools(
 
 async function syncWithLocalStorage(
   conversationID: string,
-  modelID: string,
+  modelConfig: ModelConfig,
   messageHistoryStore: MessageHistoryStore,
   client: OpenAI,
+  //  todo - reimplement with deepseek
+  // apiResponse?: ChatCompletionChunk, // Optional response for token tracking
```

Comment:
This PR shows a preview of how this will work: https://github.com/Kava-Labs/oros/blob/92220c8ad70acba29338b2ba6bac89234489e30b/src/core/context/AppContextProvider.tsx#L435